golos.publication/golos.publication.cpp

```diff
@@ -334,31 +334,54 @@ void publication::unvote(name voter, structures::mssgid message_id) {
     set_vote(voter, message_id, 0);
 }
 
-void publication::payto(name user, eosio::asset quantity, enum_t mode, std::string memo) {
+void publication::payto(std::vector<eosio::token::recipient> recipients, enum_t mode, symbol tokensymbol) {
     require_auth(_self);
-    eosio_assert(quantity.amount >= 0, "LOGIC ERROR! publication::payto: quantity.amount < 0");
-    if(quantity.amount == 0)
+    
+    print("\npayto");
+    auto publish_payout = eosio::asset(0, tokensymbol);
+    print("\n0");
+    std::vector<name> wrong_quantity;
+    print("\n1");
+
+    if (!recipients.size())
         return;
 
-    if(static_cast<payment_t>(mode) == payment_t::TOKEN) {
-        if (token::balance_exist(config::token_name, user, quantity.symbol.code())) {
-            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::code_name}, {_self, user, quantity, memo});
-        } 
-        else {
-            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::code_name}, {_self, _self, quantity, memo});
+    for (auto it = recipients.begin(); it != recipients.end(); ) {
+        print("\n2");
+        if (it->quantity.amount >= 0) {
+            print("\n3");
+            if (!token::balance_exist(config::token_name, it->to, it->quantity.symbol.code()) ||
+                !golos::vesting::balance_exist(config::vesting_name, it->to, it->quantity.symbol.code())) {
+                print("\n4");
+                if (it->quantity.amount)
+                    publish_payout.amount += it->quantity.amount;
+                it = recipients.erase(it);
```

Comment:
Don't merge payments to accounts with closed balance. We can not then determine how much and which user has been lost due to its closed balance.

Reply:
Moreover, this condition must take into account `mode`.